core/primitives/src/utils.rs

```diff
@@ -409,11 +408,9 @@ macro_rules! unwrap_or_return {
 
 /// Converts timestamp in ns into DateTime UTC time.
 pub fn from_timestamp(timestamp: u64) -> DateTime<chrono::Utc> {
+    let duration = Duration::from_nanos(timestamp);
     DateTime::from_utc(
-        NaiveDateTime::from_timestamp(
-            (timestamp / NS_IN_SECOND) as i64,
-            (timestamp % NS_IN_SECOND) as u32,
-        ),
+        NaiveDateTime::from_timestamp(duration.as_secs() as i64, duration.subsec_nanos()),
```

Comment:
The original code conversely made it clearer why the `i64` cast is fine (i.e. won't wrap around.)

Reply:
so do you suggest to keep it as it was and just add `#[allow(clippy::integer_arithmetic)]`?

Reply:
The alternative would be to use `i64::try_from().expect("infallible")` or add a comment explaining why this conversion can't wrap around.

But I think this is fine either way, just requires a little more thinking about what a sensible/valid implementation of `Duration` might be.